torch/csrc/autograd/custom_function.cpp

```diff
@@ -0,0 +1,80 @@
+#include <torch/csrc/autograd/custom_function.h>
+#include <torch/csrc/autograd/functions/accumulate_grad.h>
+
+namespace torch { namespace autograd {
+
+TORCH_API std::vector<Variable> _wrap_outputs(std::unordered_set<Variable*> &inputs,
+  std::vector<Variable*> &raw_outputs, std::unordered_set<Variable*> &non_differentiable,
+  std::vector<Variable*> &dirty_inputs, std::shared_ptr<Function> cdata) {
```

Comment:
We don't really like non-const references, so if you don't need to actually write to these, make them const. (I feel like inputs, for example, should be const, at the very least?)